features/incremental_rebuild.feature

```diff
@@ -67,6 +67,63 @@ Feature: Incremental rebuild
     And the _site directory should exist
     And I should see "Basic Site with include tag: Regenerated by Jekyll" in "_site/index.html"
 
+  Scenario: Rebuild when a data file is changed
+    Given I have a _data directory
+    And I have a "_data/colors.yml" file that contains "[red, green, blue]"
+    And I have a _data/members/core directory
+    And I have a "_data/members/core/emeritus.yml" file with content:
+      """
+      - name: John Doe
+        role: Admin
+      - name: Jane Doe
+        role: Admin
+      """
+    And I have an _includes directory
+    And I have an "_includes/about.html" file with content:
+      """
+      <ul>
+      {% for entry in site.data.members.core.emeritus %}
+        <li title="{{ entry.name }} -- {{ entry.role }}">{{ entry.name }}</li>
+      {% endfor %}
+      </ul>
+      """
+    And I have a _layouts directory
+    And I have a page layout that contains "{{ content }}\n\n{% include about.html %}"
+    And I have a home layout that contains "{{ content }}\n\nGenerated by Jekyll"
+    And I have a "_layouts/post.html" page with layout "page" that contains "{{ content }}"
+    And I have a "_layouts/static.html" page with layout "home" that contains "{{ content }}"
+    And I have an "index.html" page with layout "home" that contains "{{ site.data.colors | join: '_' }}"
+    And I have an "about.html" page with layout "page" that contains "About Us"
+    And I have a configuration file with "collections_dir" set to "collections"
+    And I have a collections/_posts directory
+    And I have the following post within the "collections" directory:
+      | title    | date       | layout  | content                      |
+      | Table    | 2009-03-26 | post    | Post with data dependency    |
+      | Wargames | 2009-03-27 | static  | Post without data dependency |
+    When I run jekyll build -IV
+    Then I should get a zero exit status
+    And the _site directory should exist
+    And I should see "red_green_blue" in "_site/index.html"
+    And I should see "John Doe -- Admin" in "_site/about.html"
```

Comment:
Should you not also see `Jane Doe` here as well?

Reply:
I've simplified the scenario by having only one entry in the YAML sequence.
i.e., `John Doe` will now be *changed* to `Jane Doe` instead (of earlier *addition* to the list) to trigger site regeneration.